homeassistant/components/lastfm/config_flow.py

```diff
@@ -0,0 +1,229 @@
+    """Config flow handler for LastFm."""
+
+    data: dict[str, Any] = {}
+
+    @staticmethod
+    @callback
+    def async_get_options_flow(
+        config_entry: ConfigEntry,
+    ) -> LastFmOptionsFlowHandler:
+        """Get the options flow for this handler."""
+        return LastFmOptionsFlowHandler(config_entry)
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Initialize user input."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            self.data = user_input.copy()
+            main_user = get_lastfm_user(
+                self.data[CONF_API_KEY], self.data[CONF_MAIN_USER]
+            )
+            errors = validate_lastfm_user(main_user)
+            if not errors:
+                return await self.async_step_friends()
+        return self.async_show_form(
+            step_id="user",
+            errors=errors,
+            description_placeholders=PLACEHOLDERS,
+            data_schema=self.add_suggested_values_to_schema(CONFIG_SCHEMA, user_input),
+        )
+
+    async def async_step_friends(
+        self, user_input: dict[str, Any] | None = None
+    ) -> FlowResult:
+        """Form to select other users and friends."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            users, errors = validate_lastfm_users(
+                self.data[CONF_API_KEY], user_input[CONF_USERS]
+            )
+            user_input[CONF_USERS] = users
+            if not errors:
+                return self.async_create_entry(
+                    title="LastFM",
+                    data={},
+                    options={
+                        CONF_API_KEY: self.data[CONF_API_KEY],
+                        CONF_MAIN_USER: self.data[CONF_MAIN_USER],
+                        CONF_USERS: [
+                            self.data[CONF_MAIN_USER],
+                            *user_input[CONF_USERS],
+                        ],
+                    },
+                )
+        try:
+            main_user = get_lastfm_user(
+                self.data[CONF_API_KEY], self.data[CONF_MAIN_USER]
+            )
+            friends = [
+                SelectOptionDict(value=friend.name, label=friend.get_name(True))
+                for friend in main_user.get_friends()
+            ]
+        except WSError:
+            friends = []
+        return self.async_show_form(
+            step_id="friends",
+            errors=errors,
+            data_schema=self.add_suggested_values_to_schema(
+                vol.Schema(
+                    {
+                        vol.Required(CONF_USERS): SelectSelector(
+                            SelectSelectorConfig(
+                                options=friends, custom_value=True, multiple=True
+                            )
+                        ),
+                    }
+                ),
+                user_input or {CONF_USERS: []},
```

Comment:
```suggestion
                user_input or {CONF_USERS: friends},
```
Should not default be `friends` which is either an empty list or the populated list with friends?